protons/app/driver.py

```diff
@@ -886,6 +918,30 @@ def attach_context(self, context):
         for force_index, force in enumerate(self.forces_to_update):
             force.updateParametersInContext(self.context)
 
+    def attach_swapper(self, swapper: Swapper, proposal: _SaltSwapProposal=None):
+        """
+        Provide a saltswapper to enable maintaining charge neutrality.
+
+        Parameters
+        ----------
+        swapper - a saltswap.Swapper object that is used for ion manipulation and bookkeeping.
+        """
+        if not isinstance(swapper, Swapper):
```

Comment:
Is it necessary to check what type `swapper` is?

Reply:
I added it to prevent confusion with a `Salinator`, which is a similar saltswap class. Perhaps I could add support for the salinator as well.

Reply:
Ah ok.

Reply:
The salinator contains a swapper, so it should be pretty simple to do if isinstance(thing, Salinator): swapper = thing.swapper.